providers/providers.go

```diff
@@ -710,3 +729,35 @@ func MustLoadSchemaFromFile(name string, path string) *resources.Schema {
 	}
 	return MustLoadSchema(name, raw)
 }
+
+// ZerologAdapter adapts the zerolog logger to the LeveledLogger interface.
+// Converts all retry logs to debug logs
+type ZerologAdapter struct {
+	logger zerolog.Logger
+}
+
+func (z *ZerologAdapter) Error(msg string, keysAndValues ...interface{}) {
+	z.logger.Debug().Fields(convertToFields(keysAndValues...)).Msg(msg)
+}
+
+func (z *ZerologAdapter) Info(msg string, keysAndValues ...interface{}) {
+	z.logger.Debug().Fields(convertToFields(keysAndValues...)).Msg(msg)
+}
+
+func (z *ZerologAdapter) Debug(msg string, keysAndValues ...interface{}) {
+	z.logger.Debug().Fields(convertToFields(keysAndValues...)).Msg(msg)
+}
+
+func (z *ZerologAdapter) Warn(msg string, keysAndValues ...interface{}) {
+	z.logger.Debug().Fields(convertToFields(keysAndValues...)).Msg(msg)
+}
+
+func convertToFields(keysAndValues ...interface{}) map[string]interface{} {
```

Comment:
Shouldn't this be an `[]interface`?

Reply:
The same in the above func signatures for the different log levels.

Reply:
We need to implement this interface 

```
type LeveledLogger interface {
	Error(msg string, keysAndValues ...interface{})
	Info(msg string, keysAndValues ...interface{})
	Debug(msg string, keysAndValues ...interface{})
	Warn(msg string, keysAndValues ...interface{})
}
```

Provided by retryablehttp - if I use `[]interface{}` instead I get 

```
panic: invalid logger type passed, must be Logger or LeveledLogger, was *providers.ZerologAdapter
```

Reply:
Thanks for checking